lib/agent.js

```diff
@@ -163,6 +165,22 @@ function PolicyAgent(options) {
     });
 }
 
+/** 
+ * Builds realm path used by /json endpoints: for details see https://backstage.forgerock.com/docs/am/6/dev-guide/#sec-rest-realms
+ */
+
+ function buildRealmPath(realm) {
+    var result = '';
+        var subrealms = realm.split('/');
+        subrealms[0] = 'root';
+        subrealms.forEach(function(r) {
+            if (r.length > 0) 
+                result += '/realms/' + r;
```

Comment:
is this backwards compatible...?

Reply:
looks like it's only valid in AM 5.0 or newer

Reply:
if AM < 5 and AM > 5 don't support the same API signature, the agent should try to use the correct format based on the AM version to keep backwards compatibility; i.e. check the server version somehow and then decide how to handle the realms

Reply:
The backwards compatible way to approach this would be to always specify the realm as a query parameter (rather than as path)

Reply:
Does it really have to be backward compatible? You could say use version up to 4.0.1 with older version of AM and new one with AM 5 or newer.

Reply:
It does, since I'm using it with AM 13 :)